build-tools/automation/azure-pipelines-oss.yaml

```diff
@@ -2,66 +2,186 @@
-
-jobs:
-- job: linux_build_package
-  displayName: Linux Build
-  pool:
-    vmImage: ubuntu-18.04
-  timeoutInMinutes: 180
-  cancelTimeoutInMinutes: 5
-  workspace:
-    clean: all
-  steps:
-  - checkout: self
-    submodules: recursive
-
-  - task: NuGetToolInstaller@1
-    displayName: 'Use NuGet 5.x'
-    inputs:
-      versionSpec: 5.x
-
-  - script: >
-      sudo apt-get install -y gnupg ca-certificates &&
-      sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF &&
-      (echo "deb https://download.mono-project.com/repo/ubuntu preview-bionic main" | sudo tee /etc/apt/sources.list.d/mono-official-preview.list) &&
-      sudo apt-get update &&
-      sudo apt-get install -y mono-devel &&
-      sudo apt-get install -y ca-certificates-mono
-    displayName: install mono preview
-
-  - script: make jenkins V=1 PREPARE_CI_PR=1 PREPARE_AUTOPROVISION=1 CONFIGURATION=$(OSSBuildConfiguration)
-    displayName: make jenkins
-
-  - script: make package-deb V=1 CONFIGURATION=$(OSSBuildConfiguration)
-    displayName: make package-deb
-
-  - script: >
-      mkdir -p $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*xamarin.android*.tar.bz2 $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*.changes $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*.dsc $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*.deb $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts
-    displayName: copy linux artifacts
-
-  - task: PublishPipelineArtifact@1
-    displayName: publish linux artifacts
-    inputs:
-      artifactName: Linux Packages
-      targetPath: $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts
-
-  - template: yaml-templates/upload-results.yaml
-    parameters:
-      configuration: $(OSSBuildConfiguration)
-      artifactName: OSS Build Results - Linux
+  XA.Build.Configuration: Release
+  NuGetArtifactName: nupkgs
+  EXTRA_MSBUILD_ARGS: /p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False
+  PREPARE_FLAGS: PREPARE_CI=1 PREPARE_CI_PR=1
+  DotNetCoreVersion: 3.1.201
+  # Version number from: https://dotnet.microsoft.com/download/dotnet-core/5.0
+  DotNetCorePreviewVersion: 5.0.100-preview.4.20227.14
+
+  BUILD_TESTS: true
+  RUN_TESTS: false
+
+stages:
+- stage: mac_stage
+  displayName: Mac
+  condition: eq(variables['EnableMacStage'], 'true')  # The variable is defined on the pipeline definition
+  jobs:
+  - job: mac_build
+    displayName: Mac Build
+    pool:
+      name: $(XA.Build.MacOSSPool)  # The variable is defined on the pipeline definition
```

Comment:
Could we move the pool names back into YAML? I think it's easier to test future changes to the pool we're using when the pool is defined in this version-able file rather than having it defined in the static pipeline settings.

Reply:
Agree that it would be better to include the pool names within the YAML file itself.  Addressed with commit bc68d05623c2b643fd3b629e1b2797ab9f7290c0